Corona-Warn-App/src/main/java/de/rki/coronawarnapp/srs/ui/typeselection/SrsTypeSelectionFragment.kt

```diff
@@ -1,17 +1,108 @@
+
     private val binding by viewBinding<FragmentSrsTypeSelectionBinding>()
 
+    @Inject lateinit var itemAdapter: SrsTypeSelectionItemAdapter
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+
+        viewModel.navigation.observe2(this) {
+            when (it) {
+                SrsTypeSelectionNavigationEvents.NavigateToCloseDialog -> showCancelDialog()
+                SrsTypeSelectionNavigationEvents.NavigateToMainScreen ->
+                    findNavController().navigate(
+                        SrsTypeSelectionFragmentDirections.actionSrsSubmissionTypeSelectionFragmentToMainFragment()
+                    )
+                SrsTypeSelectionNavigationEvents.NavigateToShareCheckins ->
+                    findNavController().navigate(
+                        SrsTypeSelectionFragmentDirections
+                            .actionSrsSubmissionTypeSelectionFragmentToSrsCheckinsFragment()
+                    )
+                SrsTypeSelectionNavigationEvents.NavigateToShareSymptoms ->
+                    findNavController().navigate(
+                        SrsTypeSelectionFragmentDirections
+                            .actionSrsSubmissionTypeSelectionFragmentToSrsSymptomsFragment()
+                    )
+            }
+        }
+
+        val backCallback = object : OnBackPressedCallback(true) {
+            override fun handleOnBackPressed() = viewModel.onCancel()
+        }
+        requireActivity().onBackPressedDispatcher.addCallback(viewLifecycleOwner, backCallback)
+
+        binding.typeList.apply {
+            layoutManager = LinearLayoutManager(requireContext())
+            adapter = itemAdapter
+        }
+
+        itemAdapter.data = typeList
+
+        itemAdapter.onItemClickListener = {
+            viewModel.selectTypeListItem(it)
+        }
+
+        binding.toolbar.setNavigationOnClickListener { viewModel.onCancel() }
+
+        viewModel.srsTestType.observe2(this) {
+            updateButtons(it)
+        }
+    }
+
+    private fun updateButtons(typeSelectionItem: SrsTypeSelectionItem) {
+
+        binding.typeList.children.forEachIndexed { i, view ->
+            (view as Button).apply {
+                handleColors(typeSelectionItem.submissionType, typeList[i].submissionType)
+            }
+        }
+
+        binding.typeSelectionNextButton.apply {
+            isEnabled = typeSelectionItem.submissionType != null
+            setOnClickListener { viewModel.onNextClicked() }
+        }
+    }
+
+    private fun Button.handleColors(submissionType: SrsSubmissionType?, state: SrsSubmissionType?) {
+        setTextColor(formatSrsTypeSelectionButtonTextStyleByState(context, submissionType, state))
+        backgroundTintList =
+            ColorStateList.valueOf(formatSrsTypeSelectionBackgroundButtonStyleByState(context, submissionType, state))
+    }
```

Comment:
I think we can do better here:
1- Use `CheckedTextView` instead of `Button` 
```xml
<CheckedTextView xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:tools="http://schemas.android.com/tools"
    android:id="@+id/type_name"
    style="@style/selectionButton"
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:layout_marginStart="12dp"
    android:layout_marginTop="@dimen/spacing_small"
    android:layout_marginEnd="12dp"
    android:backgroundTint="@color/srs_type_bg_color"
    android:textColor="@color/srs_type_text_color"
    tools:text="@string/target_button_rat_registered_no_result_text" />
```
2- Set color `selector`s for  background  and  text that changes its background based on `checked` state 
```xml
# Background 
<selector xmlns:android="http://schemas.android.com/apk/res/android">
    <item android:color="@color/colorCalendarBackgroundSelected" android:state_checked="true" />
    <item android:color="@color/colorCalendarBackgroundUnselected" />
</selector>

# Text 
<?xml version="1.0" encoding="utf-8"?>
<selector xmlns:android="http://schemas.android.com/apk/res/android">
    <item android:color="@color/colorCalendarTextSelected" android:state_checked="true" />
    <item android:color="@color/colorCalendarTextUnselected" />
</selector>
```  
3- Add checked state to the model 
```kotlin 
data class SrsTypeSelectionItem(
    val checked: Boolean = false,
    val submissionType: SrsSubmissionType?
)
```
4- Based on `SrsTypeSelectionItem` in item VH
```
You just need in `VH` binding.text.isChecked = SrsTypeSelectionItem.checked
binding.text = when(type){
.....
}
```

Reply:
see https://github.com/corona-warn-app/cwa-app-android/pull/5694